### PR TITLE
Feature/staff item requests

### PR DESCRIPTION
**Pull request by Feez2403**
Hello here is my pull request,
It add the staffing activity which allows staff to take material requests in charge.
I modified the Material request entity to have the staff name and the target event.

Please let me know if you have any suggestion

### CODE REVIEW

File: app/src/main/java/com/github/sdpteam15/polyevents/model/entity/MaterialRequest.kt [hunk 1]
```diff
@@ -15,13 +15,22 @@ data class MaterialRequest(
     val items: Map<String, Int>,
     val time: LocalDateTime?,
     val userId: String,
+    val eventId: String,
     var status: Status,
-    var adminMessage: String?
+    var adminMessage: String?,
+    var staffInChargeId: String?
 ){
     enum class Status (private val status: String) {
```
Comment (Mathematou) [inline app/src/main/java/com/github/sdpteam15/polyevents/model/entity/MaterialRequest.kt:23]: Could you add a class description?

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+import com.github.sdpteam15.polyevents.model.observable.ObservableMap
+import com.github.sdpteam15.polyevents.view.activity.staff.StaffRequestsActivity.Companion.StaffRequestStatus.*
+import com.github.sdpteam15.polyevents.view.adapter.StaffItemRequestAdapter
+
+/**
+ * Extra containing the user ID of the staff
+ */
+const val EXTRA_ID_USER_STAFF = "com.github.sdpteam15.polyevents.requests.STAFF_USER_ID"
+
+class StaffRequestsActivity : AppCompatActivity(), AdapterView.OnItemSelectedListener {
```
Comment (Mathematou) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt:29]: Comment pls

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+                zoneNameFromEventId,
+                staffId,
+                acceptMaterialDelivery,
+                cancelMaterialDelivery,
+                onMaterialDelivered
+            )
+
+        currentStatus.observe(this) {
+            refresh()
+            Log.d("YYYYYYYYYY", items.size.toString())
```
Comment (Mathematou) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt:131]: To remove Log

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+
+        currentDatabase.userDatabase!!.getListAllUsers(users)
+            .observeOnce(this) {
+                if (!it.value) {
+                    showToast("Failed to get the list of users", this)
+                }
+            }
+
+        val tempZones = ObservableList<Zone>()
+        tempZones.group { }
```
Comment (Mathematou) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt:229]: What does it do? How does it group them?

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+    override fun onItemSelected(p0: AdapterView<*>?, p1: View?, p2: Int, p3: Long) {
+        currentStatus.postValue(fromOrdinal(p2)!!)
+    }
+
+    override fun onNothingSelected(p0: AdapterView<*>?) {
+
+    }
+
+    companion object {
+        enum class StaffRequestStatus() {
```
Comment (Mathematou) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt:292]: Maybe a comment also here ?

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+                }.map { it.ordinal.toLong() }.toMutableList()
+            }
+        }
+
+        /*
+                fun allStaffStatuses(): MutableList<Long> {
+                    return StaffRequestStatus.values().flatMap { it.getItemRequestStatuses() }
+                        .toMutableList()
+                }
+        */
```
Comment (Mathematou) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt:312]: Is it useful to keep or can you remove it?

File: app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/home/StaffHomeFragment.kt [hunk 7]
```diff
@@ -1,13 +1,19 @@
...
 import androidx.fragment.app.Fragment
 import com.github.sdpteam15.polyevents.R
+import com.github.sdpteam15.polyevents.model.database.remote.Database
 import com.github.sdpteam15.polyevents.model.entity.UserRole
 import com.github.sdpteam15.polyevents.view.activity.MainActivity
+import com.github.sdpteam15.polyevents.view.activity.activityprovider.MyItemRequestsActivity
+import com.github.sdpteam15.polyevents.view.activity.staff.EXTRA_ID_USER_STAFF
+import com.github.sdpteam15.polyevents.view.activity.staff.StaffRequestsActivity
 
 class StaffHomeFragment : Fragment() {
```
Comment (Mathematou) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/home/StaffHomeFragment.kt:18]: Could you add a comment here? We probably forgot to write one before

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/activityprovider/MyItemRequestsActivity.kt [hunk 8]
```diff
@@ -127,7 +131,10 @@ class MyItemRequestsActivity : AppCompatActivity(), AdapterView.OnItemSelectedLi
         }
 
         requests.group(this, materialRequest) {
-            it.status
+            when (it.status) {
+                DELIVERING, DELIVERED, RETURN_REQUESTED, RETURNING, RETURNED -> ACCEPTED
+                else -> it.status
+            }
```
Comment (Mathematou) [inline app/src/main/java/com/github/sdpteam15/polyevents/view/activity/activityprovider/MyItemRequestsActivity.kt:137]: Nice, this way we don't have 10 millions different pages for item requests

**Review by Mathematou — APPROVED** [review]
Very nice! Just wondering why you are still using this FakeDatabase, it's so 2000... smh

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt (hunk 3 above)
Comment (Feez2403) [reply app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt:131]: Thanks, I forgot to remove it

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt (hunk 2 above)
Comment (Feez2403) [reply app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt:29]: yes

File: app/src/main/java/com/github/sdpteam15/polyevents/model/entity/MaterialRequest.kt (hunk 1 above)
Comment (Feez2403) [reply app/src/main/java/com/github/sdpteam15/polyevents/model/entity/MaterialRequest.kt:23]: yes

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt (hunk 4 above)
Comment (Feez2403) [reply app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt:229]: Thanks, I forgot to remove this

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt (hunk 5 above)
Comment (Feez2403) [reply app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt:292]: yes

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt (hunk 6 above)
Comment (Feez2403) [reply app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt:312]: Thanks, I forgot to remove it

File: app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/home/StaffHomeFragment.kt (hunk 7 above)
Comment (Feez2403) [reply app/src/main/java/com/github/sdpteam15/polyevents/view/fragments/home/StaffHomeFragment.kt:18]: True, I chenged it  and there will probably be more comments to add on activities -> last sprint

File: gradle/wrapper/gradle-wrapper.properties [hunk 9]
```diff
@@ -1,6 +1,6 @@
-#Thu Mar 04 09:21:49 CET 2021
+#Thu May 27 15:39:59 CEST 2021
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.1-all.zip
```
[the remark this answers is not in the document]
Comment (Feez2403) [reply gradle/wrapper/gradle-wrapper.properties:6]: from the gradle doc: https://docs.gradle.org/current/userguide/gradle_wrapper.html

    gradle-wrapper.properties
    A properties file responsible for configuring the Wrapper runtime behavior e.g. the Gradle version compatible with this version
I updated the gradle version, I think we can keep it on github, it's modified when we change the gradle version

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/activityprovider/MyItemRequestsActivity.kt (hunk 8 above)
Comment (Feez2403) [reply app/src/main/java/com/github/sdpteam15/polyevents/view/activity/activityprovider/MyItemRequestsActivity.kt:137]: yes, I found I think it is better for the admins who see all the requests from everybody

File: app/src/main/res/layout/activity_my_item_requests.xml [hunk 10]
```diff
@@ -4,7 +4,7 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    tools:context=".view.activity.ItemRequestActivity">
+    tools:context=".view.activity.activityprovider.MyItemRequestsActivity">
```
Comment (Rygone) [inline app/src/main/res/layout/activity_my_item_requests.xml:7]: Well done to put the right backhand

File: build.gradle [hunk 11]
```diff
@@ -6,7 +6,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.2.0'
+        classpath 'com.android.tools.build:gradle:4.1.3'
```
Comment (Rygone) [inline build.gradle:9]: Why do you have change the version to a older one?

**Review by Rygone — APPROVED** [review]

File: build.gradle (hunk 11 above)
Comment (Feez2403) [reply build.gradle:9]: I've got the following errror : This version of the Android Support plugin for IntelliJ IDEA (or Android Studio) cannot open this project, please retry with version 4.2 or newer

I tried to update everything to 4.2.1 first but it didn't work.
By searching for help on internet, I found that post https://github.com/MoIbrahim15/Android-Kotlin-MVI-CleanArchitecture/issues/3
It proposes downgrading gradle to an older version and it worked.

**Pull request merged**